app/src/main/kotlin/com/wire/android/ui/userprofile/common/UserProfileInfo.kt

```diff
@@ -0,0 +1,143 @@
+package com.wire.android.ui.userprofile.common
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.constraintlayout.compose.ConstraintLayout
+import com.wire.android.model.UserStatus
+import com.wire.android.ui.common.Icon
+import com.wire.android.ui.common.UserProfileAvatar
+import com.wire.android.ui.common.WireCircularProgressIndicator
+import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.theme.wireColorScheme
+import com.wire.android.ui.theme.wireDimensions
+import com.wire.android.ui.theme.wireTypography
+
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun UserProfileInfo(
+    isLoading: Boolean,
+    avatarAssetByteArray: ByteArray?,
+    fullName: String,
+    userName: String,
+    teamName: String?,
+    onUserProfileClick: (() -> Unit)? = null,
+    isEditable: Boolean = false,
+    onEditClick: (() -> Unit)? = null,
```

Comment:
I think it would be better to have sealed class with `onEditClick` inside the "positive" child? Like:
```
sealed class EditableState() {
    object False: EditableState()
    class True(val onEditClick: () -> Unit): EditableState()
}
```
But maybe that's just my preference 😅